src/arrow.cpp

```diff
@@ -0,0 +1,134 @@
+#include <gdf/gdf.h>
+#include <gdf/errorutils.h>
+#include <arrow/api.h>
+#include <gdf/arrow.hpp>
+#include <arrow/gpu/cuda_api.h>
+
+using namespace arrow;
+using namespace arrow::gpu;
+
+gdf_dtype arrow_dtype_to_gdf_dtype(std::shared_ptr<DataType> type){
+  gdf_dtype result;
+  switch(type->id())
+    {
+    case arrow_type::INT8:
+      result =  gdf_dtype::GDF_INT8;
+      break;
+    case arrow_type::INT16:
+      result =  gdf_dtype::GDF_INT16;
+      break;
+    case arrow_type::INT32:
+      result =  gdf_dtype::GDF_INT32;
+      break;
+    case arrow_type::INT64:
+      result =  gdf_dtype::GDF_INT64;
+      break;
+    case arrow_type::FLOAT:
+      result =  gdf_dtype::GDF_FLOAT32;
+      break;
+    case arrow_type::DOUBLE:
+      result =  gdf_dtype::GDF_FLOAT64;
+      break;
+    }
```

Comment:
Should there be a default case which raises an error?

Reply:
Yeah, we might not be able to throw for functions that run on the device, but this seems to run on the CPU.